src/lib/airtable/tables/requests.js

```diff
@@ -49,6 +66,32 @@ exports.findRequestByExternalId = async externalId => {
   }
 };
 
+// Find open requests that area already in Slack
+exports.findDeliveryNeededRequests = async () => {
```

Comment:
basically the same as `findOpenRequestsForSlack` but filters for results that are in Slack.

Reply:
Maybe this is indicating that there's a root `findOpenRequests` in our domain model that can be created and used in the two implementations?

```
let findOpenRequests = async () => { /*most of this without any slack filtering*/ }
let findDeliveryNeedeRequests = async () => { 
    let [results, err] = await findOpenRequests()
    return [results.filter(r => !requestNotInSlack(r)), err]
}

let findOpenRequestsForSlack = async () => { 
    let [results, err] = await findOpenRequests()
    return [results.filter(r => requestNotInSlack(r)), err]
}
```

Reply:
good point. i thought about it but was wary because i didn't have a way to test `findOpenRequestsForSlack`. refactoring out the two helper functions seemed safe enough. if you or someone can help me test it, we can go down the path you suggested.